### ISSUE TITLE
Theme Export: Exporting a child theme as a standalone theme creates a lot of invalid/nonsensical data in theme.json

### ISSUE THREAD

**Issue by porg** [WordPress/gutenberg#51390]
### Description

**Expectation**: The serialized `theme.json` in the new standalone theme is a proper merge of parent theme + child theme +  current user customizations (as in database).

**Actual:** Far from that. The resulting theme.json is very broken in many regards ( I identified 4 flaws, the expert probably even more). See details below.

## Reproduction

1. I had tt3dd as a child block theme of Twenty Twenty Three (aka TT3).

2. Had too many issues with child block theme. Decided to [abandon this workflow paradigm](https://github.com/WordPress/gutenberg/issues/51265#issuecomment-1584549723) and better work with a standalone theme which starts as a copy of the original TT3.

3. The child theme tt3dd may has its own mini theme.json (in my case 320 bytes, override only, had no effect anyhow) or not have any own theme.json. The outcome of the following export was the same in both situations for me! See file: 
[theme-tt3dd-export-v1.json.txt](https://github.com/WordPress/gutenberg/files/11709759/theme-tt3dd-export-v1.json.txt)

4. With the child theme tt3dd as the active theme I went to: Site Editor → Top level context menu → TOOLS → Export.

5. This exported the standalone theme which is a merge of the parent theme and the child theme into a ZIP file. Performed some renaming and adaptations in theme.css and functions.php but left the theme.json untouched. This new standalone theme (based upon TT3) now goes by the theme-slug "dd".

## Observed anomalies of the theme.json export from an active child theme

I. In the original Twenty Twenty Three theme (see: [theme-tt3.json.txt](https://github.com/WordPress/gutenberg/files/11709771/theme-tt3.json.txt)) the palette exists only once:

- settings → blocks → palette
  - slug: base

II. In [theme-tt3dd-export-v1.json.txt](https://github.com/WordPress/gutenberg/files/11709759/theme-tt3dd-export-v1.json.txt) there are several anomalies

1. palette exists twice:

  - settings → blocks → core/group → color → palette
    - slug: base
    - slug: custom-plain-inline-links -- a custom color that I created

  - settings → color → palette → slug: custom-plain-inline-links
    - slug: base -- the base color a second time
    - slug: custom-plain-inline-links -- my custom color a second time

2. settings → blocks → core/group → layout

  - Seems totally at the wrong place
  - A huge block, nearly 30% extra lines compared to original)

3. Some harmless, but at least inconsistent

  - Some of the arrays "at the end of a branch" (no more children) have there child attributes in different order in comparison to the original.

4. Some harmless, but at least inconsistent

  - Some of the arrays "at the end of a branch" (no more children) have there child attributes in different order in comparison to the original, e.g. `"fluid": { "max": "1.875rem", "min": "1.75rem" }` vs. `"fluid": { "min": "1.75rem", "max": "1.875rem" }`.
  
  - Some in the original only have slug+size vs. slug+size+name in the export, e.g. an extra name: "Extra Large" in addition to the slug: "x-large".

### Environment info

- WordPress 6.2
- Gutenberg 15.9.1

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

No

**Comment by porg** [comment]
What I did to nevertheless get a standalone theme from a former child theme:
- Duplicated a fresh TT3 theme folder.
- Copied in my custom files from `parts/` , `templates/`, `assets/css/`
- Copied in my `functions.php` and changed the parts which dealt with enqueuing parent/child CSS.
- Created a handful of edits in theme.json
- Then after some checking I was fine.

**Comment by porg** [comment]
- Possibly this had been just a result of malformed JSON.
- I will leave the issue open.
- In the coming weeks I will possibly again perform some theme duplication/forking activities, and will then observe very closely, whether I still experience this.

**Comment by SirLouen** [comment]
This report looks more like the user's diary than a real report because: The reproduction instructions are totally unclear and the only reference i've found to this `tt3dd` is [here](https://wordpress.org/support/topic/creating-child-theme-forgets-to-replace-parent-theme-references-in-templates/) (So I'm assuming is the same person)

Given by [your last answer](https://github.com/WordPress/gutenberg/issues/51390#issuecomment-1697348236), it seems that there was a problem with your original JSON

Also this might have been already been solved in #54500

I think this has been wrongly Triaged as `Needs Testing`

This report tags should be changed from `Needs Testing` either to  `Needs Author's Reply` in expectation for that supposed answer that never was received, or simply closed given that this has 2 years without answer by the author

cc @Thelmachido @Mamaduka

**Comment by porg** [comment]
Yes, your found content/reference was indeed mine.

Sorry that this bug report had been in "diary style". I could simply not further narrow it down with my knowledge (back then, now even less).

Already 2 years ago, paused development of that particular website and plan to resume it now in my upcoming summer vacation time, so hopefully I can look into it and provide feedback until end of August.

**Comment by SirLouen** [comment]
> Sorry that this bug report had been in "diary style". I could simply not further narrow it down with my knowledge (back then, now even less).

The important thing here is that you add like real and easy steps to follow. 
Like thinking that a 5-year-old could follow them. "Press here, press there". "Put this code here" "put this code there". As soon as we can reproduce the problem, we can start working on it. 
But if people take a lot of time to reproduce because the instructions are not clear, they get frustrated, and they leave the ticket without working on it.
Ping me out when you have them ready.

**Comment by porg** [comment]
Yes I'm a senior UX designer also involved in QA and software testing and know the ideal format, but for the domain subject had no clue, but also did not want to "just give up and drop it undocumented". Result was this vague report. Which ofc had little chances to be treated then.

- PR #54500 title and description mention that it avoids object duplication, which may have been the root cause for all the anomalies I had experienced.
- So when getting back to by website development project, I may revisit the possibility to not work with a theme clone, but really create a child theme, in the newest WordPress/Gutenberg with the PR integrated, and see whether the anomalies are gone then.
